File: app/schemas/booking.py
"""
schemas/booking.py — Visit booking Pydantic schemas
"""

from typing import Optional
from datetime import date, time
from pydantic import BaseModel
from enum import Enum


class BookingStatus(str, Enum):
    pending = "pending"
    confirmed = "confirmed"
    cancelled = "cancelled"
    completed = "completed"


class BookingCreate(BaseModel):
    property_id: str
    visit_date: date
    visit_time: Optional[time] = None
    message: Optional[str] = None


class BookingStatusUpdate(BaseModel):
    status: BookingStatus


class BookingResponse(BaseModel):
    id: str
    property_id: str
    buyer_id: str
    visit_date: Optional[str] = None
    visit_time: Optional[str] = None
    status: str = "pending"
    message: Optional[str] = None
    created_at: Optional[str] = None

    # Joined
    property_title: Optional[str] = None
    property_city: Optional[str] = None
    property_image: Optional[str] = None

File: android/app/src/main/kotlin/com/realestate/app/data/models/Booking.kt
package com.realestate.app.data.models

import com.google.gson.annotations.SerializedName

data class Booking(
    val id: String = "",
    @SerializedName("property_id") val propertyId: String = "",
    @SerializedName("buyer_id") val buyerId: String = "",
    @SerializedName("visit_date") val visitDate: String? = null,
    @SerializedName("visit_time") val visitTime: String? = null,
    val status: String = "pending",
    val message: String? = null,
    @SerializedName("created_at") val createdAt: String? = null,
)

data class BookingCreateRequest(
    @SerializedName("property_id") val propertyId: String,
    @SerializedName("visit_date") val visitDate: String,
    @SerializedName("visit_time") val visitTime: String,
    val message: String? = null,
)

data class BookingStatusUpdate(val status: String)
